nav2_map_server/src/map_saver/main_server_2d.cpp

```diff
@@ -14,19 +14,20 @@
 // limitations under the License.
 
 #include <memory>
-#include <stdexcept>
-#include <string>
 
 #include "nav2_map_server/map_saver.hpp"
+
 #include "rclcpp/rclcpp.hpp"
+#include "nav_msgs/msg/occupancy_grid.hpp"
 
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
 
   auto logger = rclcpp::get_logger("map_saver_server");
-  auto service_node = std::make_shared<nav2_map_server::MapSaver>();
+  auto service_node = std::make_shared<nav2_map_server::MapSaver<nav_msgs::msg::OccupancyGrid>>();
```

Comment:
Semantically, I don't really see why this shouldn't be `MapServer2D` without template arguments using the map server in the 2d directory named as such with the `template <>` removed. The actual object didn't make use of the template argument so this seems like TMI and more confusing to a later reader.